src/js/middlewares/index.js

```diff
@@ -0,0 +1,38 @@
+import { getTranslate } from 'react-localize-redux';
+import { getCountWorkflowEntityById } from 'selectors';
+
+import { setErrorsById } from 'actions';
+import { UPDATE_FIELD_VALUE } from 'actions/types';
+import cycleCountSchemes from 'schemes/cycleCountSchemes';
+import { translateWithDefaultMessage } from 'utils/Translate';
+
+const validate = (store, cycleCountId) => {
+  const { dispatch, getState } = store;
+  const state = getState();
+  const translate = translateWithDefaultMessage(getTranslate(state.localize));
+  const countEntitySchema = cycleCountSchemes.getCountEntitySchema(translate);
+  const cycleCount = getCountWorkflowEntityById(state, cycleCountId);
+  const parsedSchema = countEntitySchema.safeParse(cycleCount);
+  dispatch(setErrorsById(cycleCountId, parsedSchema.error?.format()));
+};
```

Comment:
what's the flow here? I can see the `validate` method that actually doesn't call any validation methods, it just gets some states, cycle count and the schema, and eventually set the errors, but where does the actual validation happen?